aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1634,6 +1634,9 @@ bool use_fast_accum) {
   TORCH_CHECK(mat_b.dim() == 2 || mat_b.dim() == 3, "mat_b has to be 2 or 3d");
   const bool a_is_2d = mat_a.dim() == 2;
   const bool b_is_2d = mat_b.dim() == 2;
+  if (!a_is_2d || !b_is_2d) {
+    TORCH_CHECK(mat_a.size(-1) == mat_b.size(-2), "mat_a and mat_b have to have the same trailing dimension");
```

Comment:
Should we say "contracting dimension" rather than "trailing dimension?" Would be clearer IMO